File: product-service/src/main/java/com/intexsoft/webshop/productservice/service/CategoryService.java
package com.intexsoft.webshop.productservice.service;

import com.intexsoft.webshop.productservice.dto.category.CategoryDto;
import com.intexsoft.webshop.productservice.dto.category.CategoryCreateDto;
import com.intexsoft.webshop.productservice.dto.category.CategoryUpdateDto;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CategoryService {
    CategoryDto createCategory(CategoryCreateDto categoryCreateDto);

    CategoryDto findCategoryById(Long categoryId);

    List<CategoryDto> findCategories(Pageable pageable);

    CategoryDto updateCategory(Long categoryId, CategoryUpdateDto categoryUpdateDto);

    void deleteCategoryById(Long categoryId);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/service/CategoryService.kt
package com.intexsoft.webshop.productservicekt.service

import com.intexsoft.webshop.productservicekt.dto.category.CategoryCreateDto
import com.intexsoft.webshop.productservicekt.dto.category.CategoryDto
import com.intexsoft.webshop.productservicekt.dto.category.CategoryUpdateDto
import org.springframework.data.domain.Pageable

interface CategoryService {
    fun createCategory(categoryCreateDto: CategoryCreateDto): CategoryDto
    fun findCategoryById(categoryId: Long): CategoryDto
    fun findCategories(pageable: Pageable): List<CategoryDto>
    fun updateCategory(categoryId: Long, categoryUpdateDto: CategoryUpdateDto): CategoryDto
    fun deleteCategoryById(categoryId: Long)
}